lib/dialog.rb

```diff
@@ -36,37 +40,52 @@ def invalid_message
 		@output.call "INVALID INPUT: Please enter the first letter of the colors you wish to guess\n"
 	end
 	
-	def available_colors
-		@output.call "Available Colors: #{GameState::COLORS.join(", ")}\n\n"
+	def available_colors(colors)
+		@output.call "Available Colors: #{colors.join(", ")}\n\n"
 	end
 
-	def past_guesses
-		@output.call @game_state.past_guesses.to_string
+	def past_guesses(past_guesses)
+		@output.call past_guesses.map { |g| "#{g.id}: #{g.code.join(" ")}\t red: #{g.pegs.red}  white: #{g.pegs.white}\n"}
```

Comment:
maybe pull out "`#{g.id}: #{g.code.join(" ")}\t red: #{g.pegs.red}  white: #{g.pegs.white}\n"` into a function and have it map and call the function on each item? Would be cleaner